autograder/cli/common.py

```diff
@@ -8,27 +16,89 @@
     ('skip-users', 'Skipped', 'skip'),
 ]
 
-def list_users(users, table = False):
-    if (table):
-        _list_users_table(users)
+# Set course_users to True if listing course users, False for server users.
+# An error will be raised if a user of a different type is found.
+def list_users(users, course_users, table = False, normalize = False):
+    if (course_users):
+        if (table):
+            _list_course_users_table(users)
+        else:
+            _list_course_users(users)
     else:
-        _list_users(users)
+        if (table):
+            if (normalize):
+                _list_server_users_table_normalize(users)
+            else:
+                _list_server_users_table(users)
+        else:
+            _list_server_users(users)
 
-def _list_users(users, indent = ''):
+def _list_course_users(users, indent = ''):
     for user in users:
+        if (user['type'] != "CourseType"):
+            raise ValueError("Invalid user type for listing course users: '%s'.", user['type'])
+
         print(indent + "Email:", user['email'])
         print(indent + "Name:", user['name'])
         print(indent + "Role:", user['role'])
         print(indent + "LMS ID:", user['lms-id'])
         print()
 
-def _list_users_table(users, header = True, keys = HEADERS):
-    if (header):
-        print("\t".join(keys))
+def _list_course_users_table(users, header = True, keys = COURSE_USER_HEADERS):
+    rows = []
+    for user in users:
+        if (user['type'] != "CourseType"):
+            raise ValueError("Invalid user type for listing course users: '%s'.", user['type'])
```

Comment:
You might want to be more specific with this error message. Since it's coming from _list_course_users_table, not from _list_course_users. I get why you did it because they are essentially the same error. I wanted to note it here as a small question mark in my head.